squire_lib/src/ffi/ffi_tournament.rs

```diff
@@ -637,6 +637,19 @@ impl TournamentId {
         }
     }
 
+    /// Generates a round slip for a tournament
+    /// NULL on error
+    #[no_mangle]
+    pub extern "C" fn tid_round_slips_html(self, __css: *const c_char) -> *const c_char {
+        let mut ret: String = Default::default();
+        SQUIRE_RUNTIME.get().unwrap().tournament_query(self, |t| {
+            ret = t
```

Comment:
This can be cleaned up. The point of the `tournament_query` is to perform atomic queries. So, the clone is not required. Also, the closure (that's what the `|t|` is) can return any type. So, you return the `*const c_char` directly.
```suggestion
            clone_string_c_string(&t.round_slips_html(unsafe { CStr::from_ptr(__css).to_str().unwrap() })
```

Note that `query_tournament` will return an `Option<*const c_char>` since it might not find the tournament. We should handle the `None` case like we do in other functions.